src/emoji/EmojiPickerScreen.js

```diff
@@ -0,0 +1,102 @@
+/* @flow */
+import { connect } from 'react-redux';
+
+import React, { Component } from 'react';
+import { View, Text, FlatList, StyleSheet } from 'react-native';
+import type { NavigationScreenProp } from 'react-navigation';
+
+import { emojiReactionAdd } from '../api';
+import codePointMap from './codePointMap';
+import { Screen, Input } from '../common';
+import EmojiRow from './EmojiRow';
+import getFilteredEmojiList from './getFilteredEmojiList';
+import type { GlobalState, RealmEmojiState, Auth, Dispatch } from '../types';
+import { getActiveRealmEmoji, getAuth } from '../selectors';
+import { navigateBack } from '../nav/navActions';
+import { BRAND_COLOR } from '../styles';
+
+const componentStyles = StyleSheet.create({
+  input: {
+    margin: 8,
+    borderWidth: 1,
+    borderRadius: 5,
+    borderColor: BRAND_COLOR,
+  },
+});
+
+type Props = {
+  realmEmoji: RealmEmojiState,
+  auth: Auth,
+  dispatch: Dispatch,
+  navigation: NavigationScreenProp<*> & {
+    state: {
+      params: {
+        messageId: number,
+      },
+    },
+  },
+};
+
+type State = {
+  filter: string,
+};
+
+class EmojiPickerScreen extends Component<Props, State> {
+  props: Props;
+  state: State;
+
+  state = {
+    filter: '',
+  };
+
+  handleInputChange = text => {
+    this.setState({
+      filter: text.toLowerCase(),
+    });
+  };
+
+  addReaction = item => {
+    const { auth, dispatch } = this.props;
+    const { messageId } = this.props.navigation.state.params;
+    emojiReactionAdd(auth, messageId, 'unicode_emoji', codePointMap[item], item);
+    dispatch(navigateBack());
+  };
+
+  render() {
+    const { realmEmoji } = this.props;
+    const { filter } = this.state;
+
+    const emojis = getFilteredEmojiList(filter, realmEmoji);
+
+    return (
+      <Screen title="Add Reaction">
+        <Input
```

Comment:
A quick comment, it will better to re-use the search property of the screen itself.
As now if screen is scrolled then we need to again come back to top for filtering.

Reply:
Done!